src/Main.elm

```diff
@@ -27,13 +45,21 @@ init =
 
 type Msg
     = SquareClicked Int
+    | LayTile Int
+    | CancelTile
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg model =
     case msg of
-        SquareClicked squareNumber ->
-            ( { model | selectedSquare = Just squareNumber }, Cmd.none )
+        SquareClicked squareId ->
+            ( { model | selectedSquare = Just squareId }, Cmd.none )
+
+        LayTile squareId ->
+            Debug.todo "update square with 'Tiled' state" ( model, Cmd.none )
```

Comment:
Fun feature of Elm. If you use `Debug.todo` the app will crash when it hits this point with the message you specify. But you cannot compile the app in prod mode with any `Debug` statements in your code. So it forces you to resolve them